### DISCUSSION TITLE
Filtered relation count

### DISCUSSION THREAD

**Discussion by ssukienn** [prisma/prisma#6554] [Q&A] [answered]
Hi. It is not yet a request but a question. Does Prisma provide some api to do **filtered** count on table relation? I have seen in [docs](https://www.prisma.io/docs/concepts/components/prisma-client/aggregation-grouping-summarizing#count) paragraph about `relation count` and `filtered count` (using `count` method). I wonder if there is something in the middle like defining your specific count logic inside the `include` clause with it's own `where`.

Example: 
I would like to see movie's watchers (count) but only from today or fulfilling another requirement (where) all in the same query returning list of movies (filtered in it's own way) to maintain pagination on the database level.

If it is not there, did you ever consider it or recall some similar issue to look at first? 

#EDIT: feature request ref - https://github.com/prisma/prisma/issues/6570

**Comment by ryands17** [comment]
@ssukienn 👋 

Do you want to add filters on and get the count based on that or do you want the total count and data applied with the filters?

**Reply by ssukienn** [reply]
@ryands17 if I understand you correctly, both. 

If I would describe it using normal SQL you would make your base table (movies) extra `join`  with the result of some grouped query table (watchers) where you would aggregate count `on` foreign key and **your specific filter** (like given date -> count how many watchers have seen movie in a given day, lets say watchers table has this information).

So you have your basic query with whatever filters, relations etc. and you add an extra column with aggregate value based on filtered join. This is just the solution I have thought (pseudocode):

```sql
SELECT movie_id, movie_name, ...other_movie_columns, watchers_day_count
FROM movies m
JOIN (SELECT count(*) as watchers_day_count, w.movie_id FROM watchers w WHERE w.watched_date = '<your-filter>` GROUP BY w.movie_id, w.watched_date) daily_watchers
ON m.movie_id = daily_watchers.movie_id
WHERE ...main_query_filters
```
Above is pseudocode.

`watchers_day_count` is our intended extra column from extra join which is generated with the usage of the group by aggregate using our filter in where clause

This is some specific example but probably you could generalize it to some extent. The base relation count (which already exists under `_count`) probably would be a separate join in this case because it would have a different grouping.

**Reply by ryands17** [reply, marked as answer]
Ohh, the extra filters on count not supported yet so it would be great if you could create a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) here so that we can look into this :)

**Comment by sreuter** [comment]
@ssukienn Did you end up creating a feature request for this by any chance?

**Reply by ssukienn** [reply]
Yes. It's there: https://github.com/prisma/prisma/issues/6570

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
